utils/adobe-auth.js

```diff
@@ -13,26 +13,74 @@ governing permissions and limitations under the License.
 const {
   assertImsAuthParams,
   getImsAuthProvider,
+  assertIntegrationAuthParams,
+  getIntegrationAuthProvider,
 } = require("@adobe/aio-commerce-lib-auth");
 
-const DEFAULT_IMS_SCOPES = [
-  "AdobeID",
-  "openid",
-  "read_organizations",
-  "additional_info.projectedProductContext",
-  "additional_info.roles",
-  "adobeio_api",
-  "read_client_secret",
-  "manage_client_secrets",
-  "commerce.accs",
-];
+const {
+  CommerceSdkValidationError,
+} = require("@adobe/aio-commerce-lib-core/error");
+
+const v = require("valibot");
+
+const JsonArray = v.message(
+  v.pipe(
+    v.string("Expected a JSON string"),
+    v.nonEmpty(),
+    v.parseJson(),
+    v.array(v.string()),
+  ),
+  "Oopsie",
+);
+
+const SIMPLE_STRING_REGEX = /^[\w.-]+(,\s*[\w.-]+)*$/;
+const ValidSimpleString = v.regex(
+  SIMPLE_STRING_REGEX,
+  "Scopes must be comma-separated values containing only letters, numbers, underscores, hyphens, and periods",
+);
```

Comment:
perhaps this might be better as
```suggestion
const ValidSimpleString = v.pipe(
  v.string(),
  v.regex(
    SIMPLE_STRING_REGEX,
    "Scopes must be comma-separated values containing only letters, numbers, underscores, hyphens, and periods"
  )
);
```